docs/abi-spec.rst

```diff
@@ -26,6 +26,10 @@ The first four bytes of the call data for a function call specifies the function
 first (left, high-order in big-endian) four bytes of the Keccak (SHA-3) hash of the signature of the function. The signature is defined as the canonical expression of the basic prototype, i.e.
 the function name with the parenthesised list of parameter types. Parameter types are split by a single comma - no spaces are used.
 
+.. note::
+    Return type of the function is not taken into account. In :ref:`function overloading <overload-function>` return types are not considered. The reason is to keep function call resolution context-independent.
```

Comment:
I'd clarify the link as "In Solidity's function overloading". The ABI documentation is not Solidity specific, we have just "hijacked it" and imported into our documentation :)